### ISSUE TITLE
LinkHints are duplicated on pages with plenty of links

### ISSUE THREAD

**Issue by MaxG87** [philc/vimium#3894]
**Describe the bug**
On pages that have plenty of links, such as [news.ycombinator.com](https://news.ycombinator.com), some link hints are reduced.

**To Reproduce**
1. Go to URL [news.ycombinator.com](https://news.ycombinator.com)
2. Click 'f', guess a starting letter to remove some hints and carefully look for duplicates.

In the below screenshot you can see that 'IIA' is the uppermost and the lowermost link hint.
![duplicate_link_hints](https://user-images.githubusercontent.com/5477952/131212063-41358289-8f49-4532-a0e2-3bbf4166a970.png)

**Browser and Vimium version**
Firefox: 91.0.2
OS: Debian Bullseye
Vimium: 1.67

**Comment by lydell** [comment]
Have you customized the hint chars? I know that the UI lets you put in the same char twice and that can cause issues like this.

**Comment by MaxG87** [comment]
Yes, I have customized the hint chars. I wanted to write that I checked very thoroughly to not have duplications before I created this issue. But I overlooked umlauts.

I use the following hint chars: `iuaenrtlvämqos`. When I change the `ä` to `x`, I get rid of the duplicate hints.

So the real issue is unproper handling of umlauts, and possibly other non-ASCII characters. Shall I create a new ticket for that?

**Comment by lydell** [comment]
Could it be that one hint is `IIA` and the other is `IIÄ` but _looks_ like `IIA` by “cutting off” the dots? It seems like there’s not enough space upwards for the dots. If so, a workaround could be to add some extra top padding using the “custom CSS” section.

> Shall I create a new ticket for that?

I don’t know, I’m just a random person who checks out this issue tracker sometimes :)

**Comment by gdh1995** [comment]
Um, I also think so. Though on my Win10 hint markers have enough space to show the two dots - maybe it's related with some specific font families.

**Comment by benwoodward** [comment]
I have experience this. It is reproducible if you customize your linkhint chars to `feioujlaeg`, then go to youtube.com, you should find that `GE` is duplicated.

<img width="67" alt="image" src="https://user-images.githubusercontent.com/1472981/158776926-4593f122-c155-415b-b00f-c4ac8387eb0e.png">
<img width="101" alt="image" src="https://user-images.githubusercontent.com/1472981/158776944-1b06a2cc-343a-4772-82e0-855067124921.png">

**Comment by gdh1995** [comment]
@benwoodward `Characters used for link hints` should not include duplicated characters, while in your value there're 2 `e`.

**Comment by benwoodward** [comment]
OMG I can't believe I missed that 😂. I must have been sleep deprived
